windows/shared/logging/logging-plog-windows/include/logging/windows/PlogDebugWrapper.hxx

```diff
@@ -0,0 +1,41 @@
+
+#ifndef PLOG_DEBUG_WRAPPER_HXX
+#define PLOG_DEBUG_WRAPPER_HXX
+
+#include <windows.h>
+
+#include <plog/Appenders/DebugOutputAppender.h>
+#include <plog/Appenders/DynamicAppender.h>
+#include <plog/Log.h>
+
+#include <string>
+
+namespace logging
+{
+namespace plog
+{
+
+template <typename Formatter>
+class DebugWrapperAppender : public ::plog::DynamicAppender
+{
+public:
+    /**
+     * @brief Construct a new DebugWrapperAppender object
+     * This class adds a DebugOutputApender iff the DEBUG macro is set and IsDebuggerPresent() is true
+     *  otherwise this class just does the same as the input Appender
+     * @param providerHandle
+     */
+    DebugWrapperAppender(::plog::IAppender *appender)
+    {
+        ::plog::DynamicAppender::addAppender(appender);
+        ::plog::DynamicAppender::addAppender(&m_debugAppender);
+    }
```

Comment:
Every logger implements `IAppender`, so do we need to take in an `IAppender` as input? For example, could we add this appender to the main logger that's configured with the `RollingFileAppender`? Eg.

```
using LogFormatterType = plog::TxtFormatter;
plog::RollingFileAppender<LogFormatterType> rollingFileAppender(logging::GetLogName("devicemon").c_str());
plog::DebugWrapperAppender<LogFormatterType> debugAppender;
plog::init(plog::verbose, &rollingFileAppender).addAppender(debugAppender); 
```

Reply:
Oh I didn't know that plog::init returned a logger for which you could addAppender() to. IAppender by itself doesn't provide that function but the main logger initialized by plog::init does, so we could definitely save some lines of code here.

Reply:
Ok I just pushed a change that removes the wrapper altogether because the plog::init function allows addAppender directly.